packages/core/src/middleware/koa-security-headers.ts

```diff
@@ -1,21 +1,34 @@
 import type { MiddlewareType } from 'koa';
 import helmet from 'koa-helmet';
 
+import { EnvSet, AdminApps } from '#src/env-set/index.js';
+
 /**
  * Apply security headers to the response using koa-helmet
  * @see https://cheatsheetseries.owasp.org/cheatsheets/HTTP_Headers_Cheat_Sheet.html for recommended headers
  * @see https://helmetjs.github.io/ for more details
  * @returns koa middleware
  */
 
-export default function koaSecurityHeaders<StateT, ContextT, ResponseBodyT>(): MiddlewareType<
-  StateT,
-  ContextT,
-  ResponseBodyT
-> {
+export default function koaSecurityHeaders<StateT, ContextT, ResponseBodyT>(
+  mountedApps: string[]
+): MiddlewareType<StateT, ContextT, ResponseBodyT> {
+  type Middleware = MiddlewareType<StateT, ContextT, ResponseBodyT>;
+
+  type HelmetOptions = Parameters<typeof helmet>[0];
+
+  const { isProduction, isCloud, adminUrlSet, cloudUrlSet, urlSet } = EnvSet.values;
+
+  const adminOrigins = adminUrlSet.deduplicated().map((location) => location.origin);
```

Comment:
how about add an `.origins` prop to `UrlSet`?

Reply:
cool updated